### ISSUE TITLE
Meaning of "weights" parameter in L2Loss function

### ISSUE THREAD

**Issue by aycandv** [google/trax#1355]
### Description
I have confusion about parameters of **L2Loss** function. L2Loss is implemented in trax as the following. 

I thought that it would be sufficient to get only squared error of **"model_output" and "targets"**. However the squared error is multiplied with the weights. What does that mean? Is that **"weights"** parameter the weights that are optimizable?

### Information

```
def L2Loss():
  r"""Returns a layer that computes an L2-like loss for one batch.
  The layer takes three inputs:
    - Model output from one batch, an ndarray of float-valued elements.
    - A batch of element-wise target values, which matches the shape of the
      model output.
    - A batch of weights, which matches the shape of the model output.
  The layer returns a weighted average of element-wise squared error terms
  :math:`(y_i - t_i)^2`.
  """
  def f(model_output, targets, weights):  # pylint: disable=invalid-name
    shapes.assert_same_shape(model_output, targets)
    shapes.assert_same_shape(model_output, weights)
    weighted_sse = weights * (model_output - targets)**2
    return jnp.sum(weighted_sse) / jnp.sum(weights)
  return base.Fn('L2Loss', f)
```

**Comment by friesel** [comment]
Confused the hell out of me as well. It's for your masks (so weights here are typically 0. or 1.). These are NOT the trainable weights.
